test/OrchardCore.Tests/ResourceManagement/ResourceDefinitionTests.cs

```diff
@@ -10,157 +11,223 @@ namespace OrchardCore.Tests.ResourceManagement
 {
     public class ResourceDefinitionTests
     {
-        private const string basePath = "http://host";
         private readonly ResourceManifest _resourceManifest;
 
         public ResourceDefinitionTests()
         {
             _resourceManifest = new ResourceManifest();
         }
 
-        [Fact]
-        public void GetScriptResourceWithUrl()
+        // The application path is empty if it is the host site
+        // or contains the base path of a tenant.
+        [Theory]
+        [InlineData("")]
```

Comment:
@jtkech i added the correct values for the application path here (base path)
but while I am in this piece of code what is the correct value for a virtual path? 
and I will add it as well.

leasing slash? Or without?

Reply:
/virtual/tenant

Reply:
@jtkech I updated this to add a test for the `WithBasePath` / `SetBasePath`

I think that feature, despite having some notation of being a virtualPath is disconnected from what we would actually call a virtualPath (i.e. under IIS).

Not sure how useful the `WithBasePath` actually is, but it's there, and works.

Reply:
@deanmarcussen 

That's okay it's just a question of semantics, as you want ;)

About the base path i was thinking about the httpRequest `PathBase` that may contain a virtual path (i.e. under IIS) and or a tenant prefix, the tenant prefix being a kind of virtual path, but at the end the whole being a base path.

So forget the virtual term, just use basePath, and yes better to have

    ResourceDefinition SetBasePath(string basePath)

That said in the tests, for me `/tenant` and `/virtual/tenant` are not `applicationPath`, but `basePath`, and then in `/tenant/scripts/script.js`, `/scripts/script.js` is a path

---

Okay just saw in the `ResourceDefinition` code the usage of `GetTagBuilder(..., string applicationPath, ...)` which is called somewhere by `GetHtmlContent(string appPath)` which is called by `GetHtmlContent(_options.ContentBasePath)`, and where `_options.ContentBasePath` is initialized with the httpRequest `PathBase` ;)

            options.ContentBasePath = _httpContextAccessor.HttpContext.Request.PathBase.Value;

So okay, we could say that `applicationPath` is the same as `basePath`, but here i prefer `basePath`, at least `tenantPath` as an application can have multiple tenants, but each one being a kind of an isolated app ;)

---

`https://base.com` is not a `basePath`, nor an `applicationPath`, but a full **Uri** (Url is okay), and e.g. in the full uri `https://base.com/virtual/tenant/scripts/script.js`, `/virtual/tenant` is a basePath and `/scripts/script.js` is a path

Maybe good to have a test with `/virtual/tenant`, just to do a test with a basePath having 2 segments

But as said it's just a question of terminology, so as you want ;)

Reply:
Yes, I agree @jtkech .

The terminology inside the resource definition is not quite what we are used to. So it was confusing even figuring out what they should be.

I've added `/virtual/tenant` for the two segments (makes sense)

And remove the full uri `https://base.com`. While it would work, I'm just not sure it makes sense.

Equally (and I think this is probably a different issue, but noone has ever raised it so I am ignoring it.)

I'm not sure how a base path would work in combination with a tenant prefix.

But I'm not sure exactly what you would use the `basePath` for anyway.

So I just test that it's behaviour is the **same as it is currently** so if it is being used by anyone we know there is no regression.

Reply:
Cool then, i'm fine with it

Why not checking if it starts with `file:///`, or just `file://`, or just `file:` ;)

Reply:
I don't think doing `file://` is a good idea. That would suggest that using the resource manager to load files was a good idea. Which it's not. Wouldn't work anyway the browser would block it - I would hope ;)

Reply:
@deanmarcussen No problem, that's okay ;)

Just for info, using `file:///C:/Sources/OrchardCore/src/OrchardCore.Cms.Web/App_Data/Sites/Default/Media/about-bg.jpg` works on my machine

![File Url](https://user-images.githubusercontent.com/8586360/102843376-5a946080-4409-11eb-9648-9836c0f02dd9.png)

Reply:
Yes, but try loading a script that way, on an actual site ;) 
How would I get `about-bg` to my computer. and in `C:` when I don't even have a `C:`

Reply:
> Wouldn't work anyway the browser would block it - I would hope ;)

Yes, it was just to show that the browser doesn't block an unc path starting with `file://`
It is on my local machine but i thought about something like `file://someserver/somepath`

Reply:
Ok I give in ;)

Added `file://`